internal/juju/client.go

```diff
@@ -318,6 +330,60 @@ func (sc *sharedClient) AddModel(modelName, modelUUID string, modelType model.Mo
 	}
 }
 
+// ModelStatus returns the status of the model identified by its UUID.
+func (sc *sharedClient) ModelStatus(identifier string, conn api.Connection) (*params.FullStatus, error) {
+	uuid, err := sc.ModelUUID(identifier)
+	if err != nil {
+		return nil, err
+	}
+
+	status, err := sc.getModelStatus(uuid)
+	if err != nil {
+		return sc.updateModelStatus(uuid, conn)
+	}
+
+	if time.Since(status.lastUpdate) > defaultModelStatusCacheInterval {
+		return sc.updateModelStatus(uuid, conn)
+	}
```

Comment:
I'm pondering the locking strategy here. `lastUpdate` should be checked using the a lock. I'd also hold a lock for all of the `updateModelStatus` method. It could take long enough that you're updating more often than you expect.